packages/std/src/coins.rs

```diff
@@ -206,6 +208,83 @@ impl Coins {
 
         Ok(())
     }
+
+    /// Returns an iterator over the coins.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::{coin, Coin, Coins, Uint128};
+    /// let mut coins = Coins::default();
+    /// coins.add(coin(500, "uluna")).unwrap();
+    /// coins.add(coin(1000, "uatom")).unwrap();
+    /// let mut iterator = coins.iter();
+    ///
+    /// let uatom = iterator.next().unwrap();
+    /// assert_eq!(uatom.denom, "uatom");
+    /// assert_eq!(uatom.amount.u128(), 1000);
+    ///
+    /// let uluna = iterator.next().unwrap();
+    /// assert_eq!(uluna.denom, "uluna");
+    /// assert_eq!(uluna.amount.u128(), 500);
+    ///
+    /// assert_eq!(iterator.next(), None);
+    /// ```
+    pub fn iter(&self) -> CoinsIter<'_> {
+        CoinsIter(self.0.iter())
+    }
+}
+
+impl IntoIterator for Coins {
+    type Item = Coin;
+    type IntoIter = CoinsIntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        CoinsIntoIter(self.0.into_iter())
+    }
+}
+
+impl<'a> IntoIterator for &'a Coins {
```

Comment:
Is this best practice to implement? Feels like users can accidentally get the non-owned version when they expect the owned version because they have some `&Coins` instance instead of a `Coins`.

Reply:
Yes, it's done this way for all the std collections and there's a small section about it in the [IntoIterator docs](https://doc.rust-lang.org/std/iter/index.html#iterating-by-reference).